### PR TITLE
Add CODEOWNERS, MAINTAINERS, PR template, enable DCO

### PR DESCRIPTION
**Pull request by brandond**
Uses a stripped-down version of the k3s PR template.

For MAINTAINERS - Coped from k3s, added @Oats87, fixed @briandowns username. Will push these back up to k3s if approved.

Signed-off-by: Brad Davidson <brad.davidson@rancher.com>

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Proposed changes
```
Comment (briandowns) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Do we want comments in here to help guide like we have in k3s?

File: MAINTAINERS [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# The following is the list of current K3s maintainers
```
Comment (briandowns) [inline MAINTAINERS:1]:
```suggestion
# The following is the list of current rke2 maintainers
```

File: MAINTAINERS [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# Github ID, Name, Email Address
+
+brandond, Brad Davidson, brad.davidson@rancher.com
```
Comment (briandowns) [inline MAINTAINERS:4]: Should these email be the rancher ones or the ones associated with the github account?

Comment (brandond) [reply MAINTAINERS:4]: Github uses the CODEOWNERS file for automation, I think this is more just a documentation thing? @cjellick used all of our rancher.com addresses when creating this file in k3s.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (brandond) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: I find them a bit noisy, let me try something slightly different.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+Verification
+======
+
+<!-- How can the changes be verified? Please provide whatever additional information necessary to help verify the proposed changes. -->
+
+Linked Issues
+======
+
+<!-- Link any related issues, pull-requests, or commit hashes that are relevant to this pull request. If you are opening a PR without a corresponding issue please consider creating one first, [here](https://github.com/rancher/k3s/issues). A functional example will greatly help QA with verifying/reproducing a bug or testing new features. -->
```
Comment (briandowns) [inline .github/PULL_REQUEST_TEMPLATE.md:21]:
```suggestion
<!-- Link any related issues, pull-requests, or commit hashes that are relevant to this pull request. If you are opening a PR without a corresponding issue please consider creating one first, [here](https://github.com/rancher/rke2/issues). A functional example will greatly help QA with verifying/reproducing a bug or testing new features. -->
```

Comment (brandond) [reply .github/PULL_REQUEST_TEMPLATE.md]: I tweaked that a bit, since markdown doesn't work in comments.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<!-- HTML Comments can be left in place or removed. -->
+
+Proposed changes
+======
```
Comment (cjellick) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: I find the H1 overly large. Is it possible to drop them down to a smaller heading?

Comment (brandond) [reply .github/PULL_REQUEST_TEMPLATE.md]: How's that? If we all like this I'll sync the changes back to k3s.

File: MAINTAINERS (hunk 3 above)
Comment (cjellick) [reply MAINTAINERS:4]: this is what i did in k3s...and i just yanked all the rancher addresses. It doesn't really matter from any official capacity. I would say, if someone wants to change theirs, they can. But for the purposes of getting this merged, I'm just going to roll with this

**Pull request merged**